File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteMiddle(self, head: Optional[ListNode]) -> Optional[ListNode]:
        p=head
        c=0
        while(p!=None):
            c+=1
            p=p.next
        if c==1:
            return None
        ##print(c)
        p=head
        d=c//2
        ##print(d)
        t=p
        for x in range(0,d):
            t=p
            p=p.next
        t.next=p.next
        return head
            
        

File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteMiddle(ListNode* head) {
        ListNode* p=head;
        int c{0};
        while(p)
        {
            c++;
            p=p->next;
        }
        if(c==1)
            return NULL;
        p=head;
        int dest=c/2;
        ListNode* t=p;
        for(int i=0;i<dest;i++)
        {
            t=p;
            p=p->next;
        }
        t->next=p->next;
        return head;
    }
};